src/main/java/com/zendesk/maxwell/producer/AbstractProducer.java

```diff
@@ -39,6 +39,10 @@ public AbstractProducer(MaxwellContext context) {
 
 	abstract public void push(RowMap r) throws Exception;
 
+	public boolean isDone() {
```

Comment:
should be named different, like `flushAndClose()`

Reply:
FlushAndClose is better, I have updated